pkgs/test/lib/src/runner/browser/platform.dart

```diff
@@ -185,16 +185,23 @@ class BrowserPlatform extends PlatformPlugin
       var scriptBase = htmlEscape.convert(p.basename(test));
       var link = '<link rel="x-dart-test" href="$scriptBase">';
 
-      return shelf.Response.ok('''
-        <!DOCTYPE html>
-        <html>
-        <head>
-          <title>${htmlEscape.convert(test)} Test</title>
-          $link
-          <script src="packages/test/dart.js"></script>
-        </head>
-        </html>
-      ''', headers: {'Content-Type': 'text/html'});
+      if (_config.customHtmlTemplatePath != null) {
+        var contents = File(_config.customHtmlTemplatePath).readAsStringSync();
+        var processedContents = contents.replaceFirst('{testScript}', link);
```

Comment:
Consider changing this to look for `{{testScript}}` instead of single curlies as I think that is a bit more common for template placeholders.